pytensor/tensor/subtensor.py

```diff
@@ -197,18 +230,37 @@ def get_canonical_form_slice(
     in which 0 <= start <= stop <= length and step > 0, and a flag which says
     if the resulting set of numbers needs to be reversed or not.
 
+    Returns
+    -------
+    slc
+        Slice or scalar variable.
+    direction
+        Direction to iterate the resulting elements in. (-1 or 1).
     """
     from pytensor.tensor import ge, lt, sign, switch
 
     if not isinstance(theslice, slice):
-        try:
-            value = as_index_literal(theslice)
-        except NotScalarConstantError:
-            value = theslice
-
-        value = switch(lt(value, 0), (value + length), value)
+        # Try to extract a scalar slice and return it already.
```

Comment:
This branch may be more clean if it clearly tries to not handle slices, by checking `not isinstance(theslice, slice | SliceVariable)? I don't know if `newaxis are supposed to make it to this function, I had a question below about it.

And I would call these scalar indexing instead of slices, not scalar slices?